Testing prepare merge to master

kolmanv opened the pull request:
Yaron, Gena, this branch is merge between round_clock_testing and current master.
Please check the merge is ok (only file problematic was content_data.rb).
Other were merged automatically.
After you test/fix, please merge it to master.

yarondbb committed:
Added support to create log directory if the directory does not exist

yarondbb committed:
Merge remote-tracking branch 'origin/master'

yarondbb committed:
Merge remote-tracking branch 'origin/master'

yarondbb committed:
Merge remote-tracking branch 'origin/master'

yarondbb committed:
content_data_new_design_3

genadyp committed:
spec fixed to fit new ContentData design

genadyp committed:
refactoring fix

yarondbb committed:
Developped new interface for content data and updated all usages to use this interface (2 each iterators, 1 over contents and 1 over instances)

yarondbb committed:
Merge remote-tracking branch 'origin/content_data_new_3' into content_data_new_3

genadyp committed:
validations fix with new design--

genadyp committed:
method signature changed

yarondbb committed:
fixed tests to allign with new interface of content data method:"instance_exists"

yarondbb committed:
fixed issue where inverted_index was removed due to new design

genadyp committed:
test fixed

yarondbb committed:
fixed 2 typos

yarondbb committed:
Merge remote-tracking branch 'origin/content_data_new_3' into content_data_new_3

yarondbb committed:
Merge pull request #101 from kolmanv/content_data_new_3

Content data new 3

kolmanv committed:
Merge remote-tracking branch 'origin/master' into testing_prepare_merge_to_master

Conflicts:
	lib/content_data/content_data.rb

yarondbb commented on lib/content_data/content_data.rb:
```diff
@@ -262,6 +263,11 @@ def from_file(filename)
+          Log.info "lines[i] if nil !!!, Backing filename: #{filename} to #{filename}.bad"
+          FileUtils.cp(filename, "#{filename}.bad")
+          Log.info lines[i].join("\n")
+        end
```
When content dat ais empty then number of contents and instances is 0 and 0. Is this concidered bad?
I think we need to set an empty data base and return

kolmanv replied:
This implementation considers that there are at least two lines in the code, which may be 0 and 0.
If the file has only one 0 or none at all then file is not well formatted.
Lets discuss this verbally

yarondbb commented on lib/content_data/content_data.rb:
```diff
@@ -420,6 +425,149 @@ def deep_check(instance_info)
+    end
+    is_valid
+  end
+
```
Gena, this is your part, pls see if this merge is correct

genadyp replied:
After the merge there is duplication of code: see lines 324 and 512

genadyp committed:
duplicated code was removed

yarondbb committed:
fixed issues of merge

yarondbb committed:
Fixed monitoring tests
removed redundent log tests (using tested em log4r)
Updated log. If init is not executed then init will be called on first attemp to log (same behavior as old log)

yarondbb committed:
Merge remote-tracking branch 'origin/testing_prepare_merge_to_master' into testing_prepare_merge_to_master

Conflicts:
	lib/content_data/content_data.rb

yarondbb committed:
Added flush of log in case of program abort (allign with round_clock_latest_chages)

yarondbb committed:
Updated example files with log new parameters

yarondbb committed:
Merge remote-tracking branch 'origin/round_clock_testing' into testing_prepare_merge_to_master
